### PR TITLE
Add validation check for triggers

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes

Added changes to do validation when user specify both `triggerRef`, `triggerTemplate` as part of eventlistener spec.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
User cannot specify both TriggerRef and (TriggerTemplate, TriggerBindings, Interceptors) as part of eventlistener spec.
```

### CODE REVIEW

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 1]
```diff
@@ -260,11 +260,6 @@ func Test_EventListenerValidate(t *testing.T) {
 			},
 			Spec: v1alpha1.EventListenerSpec{
 				Triggers: []v1alpha1.EventListenerTrigger{{
-					Bindings: []*v1alpha1.EventListenerBinding{{
```
Comment (dibyom) [inline pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:263]: Was this removal intentional?

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 2]
```diff
@@ -699,6 +694,27 @@ func TestEventListenerValidate_error(t *testing.T) {
...
+				Namespace: "namespace",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				Triggers: []v1alpha1.EventListenerTrigger{{
+					Template: &v1alpha1.EventListenerTemplate{
+						Ref: ptr.String("tt"),
+					},
+					TriggerRef: "triggerref",
+				}},
+				Resources: v1alpha1.Resources{
```
Comment (dibyom) [inline pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:711]: is this field required for this test?

**Review by dibyom — COMMENTED** [review]
Couple of quick nits about the test. Otherwise, looks good!

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go (hunk 2 above)
Comment (savitaashture) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:711]: Its not required 
Removed from testcase

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go (hunk 1 above)
Comment (savitaashture) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:263]: Yes because 
at a time user can specify either `triggerRef` or `triggerTemplate, triggerBinding, interceptors`

So removed `bindings` because there is `triggerRef`

Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go:263]: ahh so this was a bad test. Got it! Thanks!

**Review by dibyom — APPROVED** [review]
/lgtm
